### PR TITLE
Add possibility to disable the execve audit logs

### PR DESCRIPTION
**Pull request by QuentinBisson**
As requested by a customer, let's allow them to disable the export of execve logs

### CODE REVIEW

File: helm/fluent-logshipping-app/templates/configmap.yaml [hunk 1]
```diff
@@ -72,7 +73,9 @@ data:
         Skip_Long_Lines   ${input_skip_long_lines}
 
         threaded          on
+  {{- end }}
 
+  {{- if has "syslog" .Values.outputs.inputLogTypes }}
```
Comment (hervenicol) [inline helm/fluent-logshipping-app/templates/configmap.yaml:78]: This is not related to `execve`, right?

File: helm/fluent-logshipping-app/templates/configmap.yaml [hunk 2]
```diff
@@ -48,10 +48,11 @@ data:
     {{- if has "syslog" .Values.outputs.inputLogTypes }}
     @INCLUDE inputs-syslog.conf
     {{- end }}
-    {{- if or (has "audit" .Values.outputs.inputLogTypes) (has "sshd" .Values.outputs.inputLogTypes) }}
+    {{- if or (has "audit" .Values.outputs.inputLogTypes) (has "sshd" .Values.outputs.inputLogTypes) (has "execve" .Values.outputs.inputLogTypes) }}
     @INCLUDE inputs-auditing.conf
     {{- end }}
 
+  {{- if has "containers" .Values.outputs.inputLogTypes }}
```
Comment (hervenicol) [inline helm/fluent-logshipping-app/templates/configmap.yaml:55]: This is not related to `execve`, right?

File: helm/fluent-logshipping-app/templates/configmap.yaml [hunk 3]
```diff
@@ -82,7 +85,9 @@ data:
         Mem_Buf_Limit     ${input_memory_buffer_limit}
         Strip_Underscores on
         threaded          on
+  {{- end }}
 
+  {{- if or (has "audit" .Values.outputs.inputLogTypes) (has "sshd" .Values.outputs.inputLogTypes) (has "execve" .Values.outputs.inputLogTypes) }}
```
Comment (hervenicol) [inline helm/fluent-logshipping-app/templates/configmap.yaml:90]: So, the idea is to create this file only in cases where it will be included?
Then, the `{{end}}` should be after all the inputs are defined?

File: helm/fluent-logshipping-app/templates/configmap.yaml (hunk 2 above)
Comment (QuentinBisson) [reply helm/fluent-logshipping-app/templates/configmap.yaml:55]: No it is not, sorry, this is indeed a refactoring to prevent rendering this part as it is unused and caused me confusion when debugging

File: helm/fluent-logshipping-app/templates/configmap.yaml [hunk 4]
```diff
@@ -97,7 +102,9 @@ data:
         Systemd_Filter    SYSLOG_IDENTIFIER=sshd
         Systemd_Filter    _SYSTEMD_UNIT=sshd.service
         threaded          on
+    {{- end }}
```
Comment (QuentinBisson) [inline helm/fluent-logshipping-app/templates/configmap.yaml:105]: This end is related to line 92

File: helm/fluent-logshipping-app/templates/configmap.yaml [hunk 5]
```diff
@@ -126,11 +133,13 @@ data:
         Refresh_Interval  ${input_refresh_interval_in_seconds}
         Skip_Long_Lines   ${input_skip_long_lines}
     {{- end }}
+  {{- end }}
```
Comment (QuentinBisson) [inline helm/fluent-logshipping-app/templates/configmap.yaml:136]: This end is related to the big if

**Comment by hervenicol** [comment]
If I understand it right, `execve` input was disabled when we were disabling `sshd` input. And now they're 2 different configs.
It means some users could currently have disabled `sshd` input, and will receive all `execve` logs once this PR is merged?

File: helm/fluent-logshipping-app/templates/configmap.yaml (hunk 4 above)
Comment (hervenicol) [reply helm/fluent-logshipping-app/templates/configmap.yaml:105]: oh, my bad, it's because of the `sshd` / `execve` split! I see it now! :+1:

**Comment by QuentinBisson** [comment]
So right now, when disabling sshd, we disabled both sshd and execve and yes they are 2 different configs

Once this PR is merged, customers that enabled sshd logs will receive only sshd logs and not sshd and execve logs.

This is deployed in only 1 place right now so once it's released I'll add the execve log property

File: helm/fluent-logshipping-app/templates/configmap.yaml (hunk 4 above)
Comment (QuentinBisson) [reply helm/fluent-logshipping-app/templates/configmap.yaml:105]: :)

**Comment by hervenicol** [comment]
All right, then LGTM.

**Review by hervenicol — APPROVED** [review]
Allr
